Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
+        {
+            var lossForCurrentPerson = 0M;
+
+            for (int j = 0; j < data.Length - 1; j++)
+            {
+                var isVallid = decimal.TryParse(data[j], out decimal amount1);
+                
+                var isValid = decimal.TryParse(data[j + 1], out decimal amount2);
+                lossForCurrentPerson = amount1 - amount2;
+            }
+
+            return lossForCurrentPerson; 
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestBalance = decimal.MinValue;
+            var RichestPerson = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var isValid = decimal.TryParse(currentPersonData[currentPersonData.Length - 1], out decimal highestAmountOfPerson);
+
+                if (!isValid)
+                {
+                    return "N/A.";
+                }
+
+                HighestBalance = FindRichestPerson(peopleAndBalances, HighestBalance, RichestPerson, i, currentPersonData, highestAmountOfPerson);
+
+            }
+
+            var answer = $"{RichestPerson.ToString()} is the richest person. ¤{HighestBalance}.";
+
+            return answer;
+
+
+        }
+
+        private static decimal FindRichestPerson(string[] peopleAndBalances, decimal HighestBalance, StringBuilder RichestPerson, int i, string[] currentPersonData, decimal highestAmountOfPerson)
+        {
+            if (highestAmountOfPerson > HighestBalance)
+            {
+                HighestBalance = highestAmountOfPerson;
+                var currentOne = RichestPerson.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    RichestPerson.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    RichestPerson.Replace(currentOne, currentPersonData[0]);
+                }
+
+
+            }
+            else if (highestAmountOfPerson == HighestBalance)
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    RichestPerson.Append(" and ");
+                }
+                else
+                {
+                    RichestPerson.Append(", ");
+                }
+
+                RichestPerson.Append(currentPersonData[0]);
+            }
+
+            return HighestBalance;
+        }
+
+        private static void FindMorePeopleWhicAreEvenPoor(string[] peopleAndBalances, decimal HighestBalance, StringBuilder RichestPerson, int i, string[] currentPersonData, decimal highestAmountOfPerson)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'FindMorePeopleWhicAreEvenPoor'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)

Reply:
done